dashboards-observability/public/components/explorer/explorer.tsx

```diff
@@ -822,6 +900,144 @@ export const Explorer = ({
     }
   };
 
+  const onToggleChange = (e: { target: { checked: boolean | ((prevState: boolean) => boolean); }; }) => {
+    setLiveTailToggle(e.target.checked);
+    setIsLiveTailPopoverOpen(!isLiveTailPopoverOpen);
+  };
+
+  const wrappedPopoverButton = useMemo(() => {
+    return (
+      <EuiButtonToggle
+        label={liveTailNameRef.current}
+        iconType="arrowDown"
+        isIconOnly={isLiveTailOn ? false : true}
+        iconSide="right"
+        onClick={() => setIsLiveTailPopoverOpen(!isLiveTailPopoverOpen)}
+        onChange={onToggleChange}
+        data-test-subj="eventLiveTail"
+      />
+    );
+  }, [isLiveTailPopoverOpen, liveTailToggle, onToggleChange, isLiveTailOn]);
+
+  const sleep = (milliseconds: number | undefined) => {
+    return new Promise(resolve => setTimeout(resolve, milliseconds))
+  }
+
+  const liveTailLoop = async (name: string, startTime: string, endTime: string, delayTime: number) => {
+    setLiveTailName(name)
+    setLiveTailTabId(curSelectedTabId.current);
+    setIsLiveTailOn(true);
+    setToast("Live tail On",'success');
+    setIsLiveTailPopoverOpen(false);
+    await sleep(2000);
+    let curLiveTailname = liveTailNameRef.current;
+    while (isLiveTailOnRef.current === true && curLiveTailname === liveTailNameRef.current){
+      handleLiveTailSearch(startTime, endTime);
+        if (liveTailTabIdRef.current !== curSelectedTabId.current) {
+          setIsLiveTailOn(!isLiveTailOnRef.current);
+          isLiveTailOnRef.current = false;
+          setLiveTailName("")
+        }
+        await sleep(delayTime);
```

Comment:
just a note, might be better to use a single `setInterval` and clear interval when loop not needed